### PR TITLE
[OPT] FlashAttention && ModelParallel

### PR DESCRIPTION
**Pull request by ForFishes**
### PR types

Performance optimization
### PR changes

Others
### Describe

opt gpt performace

### CODE REVIEW

File: paddle/phi/kernels/gpu/flash_attn_kernel.cu [hunk 1]
```diff
@@ -75,11 +75,15 @@ void FlashAttnRawKernel(const Context& ctx,
...
   auto seed_offset_pair = gen->IncrementOffset(inc);
+
   uint64_t seed = seed_offset_pair.first;
   uint64_t offset = seed_offset_pair.second;
 
-  std::vector<int64_t> seed_offset_vec{int64_t(seed), int64_t(offset)};
-  phi::TensorFromVector<int64_t>(seed_offset_vec, ctx, seed_offset);
+  paddle::platform::CPUPlace cpu_place;
+  seed_offset->Resize({2});
+  auto* seed_offset_data = seed_offset->mutable_data<uint64_t>(cpu_place);
```
Comment (sneaxiy) [inline paddle/phi/kernels/gpu/flash_attn_kernel.cu]: Use `auto* seed_offset_data = ctx.template HostAlloc<uint64_t>(seed_offset)` instead.

File: python/paddle/distributed/fleet/layers/mpu/mp_layers.py [hunk 2]
```diff
@@ -329,6 +330,18 @@ def forward(self, x):
         return output
 
 
+class MPScale(PyLayer):
+    @staticmethod
+    def forward(ctx, x, mp_degree):
+        out = paddle.scale(x, 1.0 / mp_degree)
+        ctx.mp_degree = mp_degree
```
Comment (sneaxiy) [inline python/paddle/distributed/fleet/layers/mpu/mp_layers.py]: Remove this line.

File: paddle/phi/kernels/gpu/flash_attn_kernel.cu (hunk 1 above)
Comment (ForFishes) [reply paddle/phi/kernels/gpu/flash_attn_kernel.cu]: done, thx

File: python/paddle/distributed/fleet/layers/mpu/mp_layers.py (hunk 2 above)
Comment (ForFishes) [reply python/paddle/distributed/fleet/layers/mpu/mp_layers.py]: done.

**Review by sneaxiy — APPROVED** [review]

**Pull request merged**